### DISCUSSION TITLE
Visual Studio 2019 IDE

### DISCUSSION THREAD

**Discussion by CryptxzzX** [community/community#6808] [Copilot Conversations] [answered]
Would love to be able to use this with Visual Studio 2019 IDE

I primarily  code in C# and use WinForms which unfortunately isn't doable in VSCode.

Other then that I'm having alot of fun!

**Comment by hsenag** [comment, marked as answer]
Hi, thanks very much for the feedback. We do intend to expand the range of editors we support over time.

**Reply by CryptxzzX** [reply]
That's great to hear! Can't wait to see what's to come for  Copilot.

**Comment by frozenprakash** [comment]
Kindly do provide the support for IDE asap,
The number of users on Visual studio IDE is quite high, so a marketplace extension for VS2019 or VS2022 as majority of developers are there with complex projects which will also be a great training model for GitHub CoPilot.

**Reply by CryptxzzX** [reply]
I 100% agree.

**Comment by catalin-ifacts** [comment]
Most of us C# devs, use Visual Studio, not VSCode.

**Reply by Kek5chen** [reply]
same for c++ 👍

**Comment by MulleDK19** [comment]
Considering Copilot is a Microsoft project, it's absolutely baffling that VS isn't already the number one supported IDE....?
